liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java

```diff
@@ -540,6 +545,33 @@ protected void copyConfigFiles() throws Exception {
         copyDependencies();
     }
 
+    /**
+     * Merges envProps with special properties found in the install (target) server.env.  We return a clone/copy of
+     * envProps, to which any of a list of special properties found in server.env have been added.  We give precedence
+     * to properties already in envProps.
+     */
+    private Map<String, String> mergeSpecialPropsFromInstallServerEnvIfAbsent(Map<String, String> envProps) throws IOException {
+
+        String[] specialProps = { "keystore_password" };
+
+        Map<String, String> mergedProps = new HashMap<String,String>();
+        
+        // Clone to avoid side effects 
+        for (String key : envProps.keySet()) {
+            mergedProps.put(key, envProps.get(key));
+        }
+
+        // From install (target) dir
+        File serverEnv = new File(serverDirectory, "server.env");
+        Map<String, String> serverEnvProps = convertServerEnvToProperties(serverEnv);
+        
+        for (String propertyName : specialProps) {
+            mergedProps.putIfAbsent(propertyName, serverEnvProps.get(propertyName));
```

Comment:
This line of code seems to presume that "keystore_password" will always exist in the target server.env, which is not the case. So I would change this to:

`if (serverEnvProps.containsKey(propertyName)) { mergedProps.putIfAbsent(...); }`

Reply:
Thx